### DISCUSSION TITLE
In the config.csv - is it possible to use variables from secrets in a KeyVault

### DISCUSSION THREAD

**Discussion by ken-vetter-absg** [Azure-Player/azure.datafactory.tools#392] [Q&A]
I have a key vault with secrets for values that will be used in the ADF-Publish pipleine.
With a variable group (with secrets linked from a KV), linked to the ADF-Publish release pipeline, I can use those variables, but I have to create a mapping in the Publish ADF task so the task is aware of them.  I want to make a direct link from the config.csv to the variable in the KV.  Example:
I have a Linked Service in my ADF that is linked to a Key Vault.  To deploy my ADF from test to stage, the URL to the KV for stage needs to updated.
Config entry looks like this:
linkedService,<LSName>,typeProperties.baseUrl,"$Env:KVURL"

When $ENV:KVURL references a pipeline variable, the value for URL is updated in my stage ADF
When $ENV:KVURL is trying to use a Key Vault value with secret name "KVURL" the value in Stage is blank.
I am using the Azure Key Vault Task to pull the secrets into the agent (pre-job) and set to share for all tasks in the job, but it seems the publish task isn't able to recognize them.

One other note, when I did have the secrets linked to the pipeline through a Variable Library (linked to the KV), I used the Environment Variables section in the Publish task to map the variables I am using in the config.sys to the variable linked in the variable group.  This works, but I don't want to have the added step of every time I add a secret to the KV for ADF deployments, I have to make a manual entry in the task for a local mapping.

How do I make the entry in the config.sys recognize key vault secrets?

**Comment by NowinskiK** [comment]
Secure variables are not available in a task's session by default primarily due to security concerns and design considerations.
Their absence is a deliberate security measure to minimize the risk of unauthorized access or exposure of sensitive information. It underscores the importance of carefully managing and controlling access to sensitive data within your pipelines.
That's why you must use the "Environment Variables" section to map those of secure variables which you want to expose as environment variables in task's session. 
To avoid this - don't store every single value in AKV - most of them are not sensitive information (like AKV name/URL).
